src/api/java/com/minecolonies/api/research/AbstractResearchProvider.java

```diff
@@ -26,16 +27,16 @@
  */
 public abstract class AbstractResearchProvider implements IDataProvider
 {
-    public static final String ASSETS_DIR = "assets/" + Constants.MOD_ID + "/researches";
-    private final DataGenerator generator;
+    public static final String ASSETS_DIR = "data/" + Constants.MOD_ID + "/researches";
+    protected final GatherDataEvent event;
 
     /**
      * The abstract variant of a ResearchProvider, to register to fires during runData.
-     * @param generator  The DataGenerator to store the files.
+     * @param event  The GatherDataEvent to store the files.
      */
-    public AbstractResearchProvider(final DataGenerator generator)
+    public AbstractResearchProvider(final GatherDataEvent event)
```

Comment:
It's bad style to pass the event into the provider.  Pass the `DataGenerator` and `ExistingFileHelper` as separate parameters instead.